src/Components/pageComponents/auth/login/index.tsx

```diff
@@ -0,0 +1,312 @@
+                setIsLoading(true);
+                const { email, password } = values;
+                if (rememberMe) {
+                    const expirationDate = new Date();
+                    expirationDate.setDate(expirationDate.getDate() + 30);
+                    localStorage.setItem('email', email);
+                    localStorage.setItem('password', password);
+                    localStorage.setItem('rememberMe', 'true');
+                } else {
+                    localStorage.removeItem('email');
+                    localStorage.removeItem('password');
+                    localStorage.removeItem('rememberMe');
+                }
+
+                // manual login
+                try {
+                    const response = await sendRequest('user/login', {
+                        method: 'POST',
+                        data: { email, password },
+                    });
+
+                    setIsLoading(false);
+                    if (response.status === 200) {
+                        const userDetails = {
+                            name: response?.data?.userData?.fullName,
+                            email: response?.data?.userData?.email,
+                        };
+                        if (response?.data?.userData) {
+                            localStorage.setItem('userData', JSON.stringify(response.data?.userData));
+                        }
+                        updateUserInfo(userDetails);
+                        localStorage.setItem('jwtToken', response?.data?.userData?.token);
+
+                        handleRedirect(response?.data?.userData?.token);
+                    } else {
+                        setError('Invalid email or password');
+                    }
+                } catch (error: any) {
+                    console.log('Error in Login API => ', error);
+                    setIsLoading(false);
+                    setError('Login Failed, Please try again later');
+                } finally {
+                    setSubmitting(false);
+                }
+            },
+        });
+
+    const handleRedirect = (token: any) => {
+        if (token) {
+            const decodedToken: any = decodeJWt(token);
+            console.log('tokennnn', decodedToken.role.role);
+            if (decodedToken.role.role === 'admin') {
+                router.push('/adminDashboard/room');
+            } else if (decodedToken.role.role === 'user') {
+                router.push('/userDashboard/tournament');
+                // router.push(configData.web.cominSoonUrl)
+            } else {
+                router.push('/spectatorDashboard');
+            }
+        } else {
+            router.push('/auth/401');
+        }
+    };
+
+    useEffect(() => {
+        const storedEmail = localStorage.getItem('email');
+        const storedPassword = localStorage.getItem('password');
+        if (storedEmail) {
+            setFieldValue('email', storedEmail);
+        }
```

Comment:
check this useEffect again

Reply:
what are the changes you want to implement in this useeffect